### PR TITLE
chore: improve imports for configurations and CredentialProvider

### LINKED ISSUE #284: improve the import situation for Configurations
Multiple devs reported issues trying to figure out how to import the configuration objects.

I think my preference would be to be able to do `from momento import configurations` and then in the constructor I would do `configurations.Laptop.v1()`, but if there is a more pythonic way to handle it that's fine.  At least needs to be more discoverable, perhaps by adding import statement to the docstring example code.

### PR DESCRIPTION
**Pull request by malandis**
Rolls up configurations and CredentialProvider to top level `momento`
package. Updates documentation of inline examples.

Closes #284

### CODE REVIEW

File: src/momento/__init__.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import logging
 
+from .auth import CredentialProvider
+from .config import configurations
```
Comment (schwern) [inline src/momento/__init__.py:4]: This is an oddball. Can it be Configurations?

**Review by schwern — COMMENTED** [review]
This is so much better. I wonder if we can do something similar with responses?

File: src/momento/__init__.py (hunk 1 above)
Comment (malandis) [reply src/momento/__init__.py:4]: I hear you. We'd have to define an outer class in that module that contains all the pre-built configurations

Comment (malandis) [reply src/momento/__init__.py]: Refactored in [afcef14](https://github.com/momentohq/client-sdk-python/pull/295/commits/afcef14a8c9787d05727dffb3131c47679c8fa78)

**Review by schwern — APPROVED** [review]

**Review by nand4011 — APPROVED** [review]
It's easy to miss the examples in the docs. Well caught.

**Comment by malandis** [comment]
> This is so much better. I wonder if we can do something similar with responses?

I'm conflicted between making the responses easy to discover -- ie at the top level API -- vs grouping them nicely under cohesive package -- `momento.responses`

**Pull request merged**